pyanaconda/ui/lib/disks.py

```diff
@@ -92,3 +92,33 @@ def applyDiskSelection(storage, data, use_names):
 
     data.ignoredisk.onlyuse = onlyuse
     data.clearpart.drives = use_names[:]
+
+def getRelatedDisks(storage, disk):
+    """ Return a list of disks related to disk by way of container devices. """
+    return set(d for dep in storage.deviceDeps(disk) for d in dep.disks)
+
+def checkDiskSelection(storage, all_disks, selected_disks):
+    """ Return a list of errors related to a proposed disk selection.
+
+        :param :class:`blivet.Blivet` storage: storage data
+        :param all_disks: all known disks
+        :type all_disks: list of :class:`blivet.devices.StorageDevice`
+        :param selected_disks: proposed selected disks
+        :type selected_disks: list of :class:`blivet.devices.StorageDevice`
+        :returns: a list of error messages
+        :rtype: list of str
+    """
+    errors = []
+    for selected in all_disks:
+        if selected.name not in selected_disks:
+            continue
```

Comment:
I don't understand the reason for iterating over ``all_disks`` and then skipping all of them that are not in ``selected_disks``. Could there ever be a situation that there will be a disk in ``selected_disks`` that is not in ``all_disks``? If not we can just iterate over ``selected_disks`` making ``all_disks`` an unused/unneeded argument that could be dropped.